server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateIntervalWrapper.java

```diff
@@ -69,28 +66,30 @@ public void writeTo(StreamOutput out) throws IOException {
         public String value() {
             return name().toLowerCase(Locale.ROOT);
         }
+
+        public boolean isValid() {
+            // I'm being a little cheeky here and just reusing the name for signaling invlaid choices too
+            return this.preferredName != null;
+        }
+
+        public String getPreferredName() {
+            if (preferredName == null) {
+                throw new IllegalStateException("Invalid use of legacy date histogram interval");
+            }
+            return preferredName;
+        }
+
+        private String preferredName;
+        IntervalTypeEnum(String preferredName) {
+            this.preferredName = preferredName;
+        }
     }
 
     private DateHistogramInterval dateHistogramInterval;
     private IntervalTypeEnum intervalType = IntervalTypeEnum.NONE;
 
     public static <T extends DateIntervalConsumer> void declareIntervalFields(ObjectParser<T, String> parser) {
 
-        // NOTE: this field is deprecated and will be removed
-        parser.declareField((wrapper, interval) -> {
-            if (interval instanceof Long) {
-                wrapper.interval((long) interval);
-            } else {
-                wrapper.dateHistogramInterval((DateHistogramInterval) interval);
-            }
-        }, p -> {
-            if (p.currentToken() == XContentParser.Token.VALUE_NUMBER) {
-                return p.longValue();
-            } else {
-                return new DateHistogramInterval(p.text());
-            }
-        }, Histogram.INTERVAL_FIELD, ObjectParser.ValueType.LONG);
```

Comment:
to support compatibility you can set `.forRestApiVersion(equalTo(RestApiVersion.V_7)` on Histogram.INTERVAL_FIELD
this will only parse the field for v7 requests

Reply:
yup, I'm working on that now.  It needs to be slightly more clever, because I don't want to keep the field itself in 8.0.  Instead we can make a reasonable guess as to how to turn the 7.x version into one of the fields we're replacing it with in 8.0.  I'll ping you when I have a draft ready